### PR TITLE
Histogram bounds validation

### LINKED ISSUE #2566: Histogram bounds must be validated
Similar to instrument/view name, we need to validate the custom bounds (no duplicates, increasing order etc.), if provided.
If we can fail-fast, do that. - done in #2573 
Else log and move on.

### PR DESCRIPTION
**Pull request by mic-max**
Fixes #2566

## Changes

Throws `ArgumentException` when invalid histogram bounds are provided.
The bounds must be in ascending order and contain distinct values.

For significant contributions please make sure you have completed the following items:

* [x] `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: src/OpenTelemetry/Metrics/MeterProviderSdk.cs [hunk 1]
```diff
@@ -125,6 +125,15 @@ internal sealed class MeterProviderSdk : MeterProvider
                         var metricStreamConfig = viewConfig(instrument);
                         if (metricStreamConfig != null)
                         {
+                            if (metricStreamConfig is HistogramConfiguration histogramConfiguration)
+                            {
+                                // Validate the histogram bounds
+                                if (!this.IsSortedAndDistinct(histogramConfiguration.BucketBounds))
```
Comment (mic-max) [inline src/OpenTelemetry/Metrics/MeterProviderSdk.cs:131]: Is there a way I can detect that this value is not the default?
In that case the validation check could be omitted.

File: test/OpenTelemetry.Tests/Metrics/MetricViewTests.cs [hunk 2]
```diff
@@ -78,6 +79,29 @@ public void AddViewWithInvalidNameThrowsArgumentException(string viewNewName)
...
+        }
+
+        [Theory]
+        [MemberData(nameof(MetricsTestData.InvalidHistogramBounds), MemberType = typeof(MetricsTestData))]
+        public void AddViewWithInvalidHistogramBoundsThrowsArgumentException(int index, double[] bounds)
+        {
+            using var meter = new Meter($"{Utils.GetCurrentMethodName()}.{index}");
+
+            var ex = Assert.Throws<ArgumentException>(() => Sdk.CreateMeterProviderBuilder()
+                .AddMeter(meter.Name)
```
Comment (mic-max) [inline test/OpenTelemetry.Tests/Metrics/MetricViewTests.cs:98]: Is this line required?

File: test/OpenTelemetry.Tests/Metrics/MetricViewTests.cs [hunk 3]
```diff
@@ -78,6 +79,29 @@ public void AddViewWithInvalidNameThrowsArgumentException(string viewNewName)
...
+        public void AddViewWithValidHistogramBounds(int index, double[] bounds)
+        {
+            Sdk.CreateMeterProviderBuilder()
+                .AddView("name1", new HistogramConfiguration { BucketBounds = bounds })
+                .Build();
+        }
+
+        [Theory]
+        [MemberData(nameof(MetricsTestData.InvalidHistogramBounds), MemberType = typeof(MetricsTestData))]
+        public void AddViewWithInvalidHistogramBoundsThrowsArgumentException(int index, double[] bounds)
```
Comment (mic-max) [inline test/OpenTelemetry.Tests/Metrics/MetricViewTests.cs:93]: This isn't throwing the exception currently. @cijothomas do you understand why not?

File: test/OpenTelemetry.Tests/Metrics/MetricsTestData.cs [hunk 4]
```diff
@@ -40,5 +40,20 @@ public static IEnumerable<object[]> ValidInstrumentNames
                     new object[] { "my_metric2" },
                     new object[] { new string('m', 63) },
            };
+
+        public static IEnumerable<object[]> ValidHistogramBounds
+            => new List<object[]>
+            {
+                new object[] { 0, new double[] { } },
+                new object[] { 1, new double[] { 10 } },
```
Comment (mic-max) [inline test/OpenTelemetry.Tests/Metrics/MetricsTestData.cs:48]: Are empty and single element bounds allowed?

File: src/OpenTelemetry/Metrics/MeterProviderSdk.cs [hunk 5]
```diff
@@ -125,6 +125,15 @@ internal sealed class MeterProviderSdk : MeterProvider
...
+                                {
+                                    throw new ArgumentException("Must be in ascending order with distinct values", $"{histogramConfiguration}.{histogramConfiguration.BucketBounds}");
```
Comment (cijothomas) [inline src/OpenTelemetry/Metrics/MeterProviderSdk.cs:133]: we shouldn't throw here. as this code is occuring after SDK has successfully initialized.

**Pull request closed without merging**